static/src/javascripts/projects/common/modules/discussion/activity-stream.js

```diff
@@ -59,5 +97,20 @@ define([
         });
     }
 
+    function selectTab(streamType, fallback) {
+        var from = $('.tabs__tab--selected');
+        var to = $('a[data-stream-type=' + streamType + ']');
+
+        if (to.length == 0) {
+            to = $('a[data-stream-type=' + fallback + ']');
+        }
+
+        from.removeClass('tabs__tab--selected');
+        var link = $('a', from);
+        link.blur();
+
+        bonzo(to).parent().addClass('tabs__tab--selected');
```

Comment:
I think we could avoid doing the DOM jumping here by adding the stream type to a class on the `li`, that way you can target it directly. Minor though, don't know if the mark-up for this is somewhere horrible.